src/main/kotlin/net/weavemc/loader/api/event/EventTypes.kt

```diff
@@ -13,28 +13,72 @@ import net.minecraft.world.World
 import org.lwjgl.input.Keyboard
 import org.lwjgl.input.Mouse
 
+/**
+ * This is the base class for all events provided by the Weave Loader.
+ */
 public abstract class Event
+
+/**
+ * This is the base class for all *cancellable* events provided by the
+ * Weave Loader, extending [Event].
+ */
 public abstract class CancellableEvent : Event() {
+
+    /**
+     * This field defines whether the event is cancelled or not. Any mod can cancel and
+     * un-cancel an event. What an event does when cancelled is event-specific, and noted in
+     * that event's documentation.
+     */
     @get:JvmName("isCancelled")
     public var cancelled: Boolean = false
 
 }
 
+/**
+ * This is the base class for a Tick Event. Tick Events can be Pre and Post, but you will never
+ * receive a base Tick Event. These are each called every game tick, or roughly every `50/timer` milliseconds.
+ * The game's timer speed will never change (unless modified by cheats), so it is safe to assume that it is `1`.
+ */
 public sealed class TickEvent : Event() {
+
+    /**
+     * Pre Tick Events are called right before the game sends that tick's **position, rotation,
+     * and onGround** updates to the server, and as such, they can be modified here. **If you
+     * modify any of them here, it might be a good idea to return them to their original value on
+     * the following [Post Tick Event][Post]**.
+     */
     public object Pre : TickEvent()
+
+    /**
+     * Post Tick Events are called right after the game sends that tick's **position, rotation,
```

Comment:
Same with this, keep it short and concise. No need to document the good use cases for it, as there are many use cases. On top of that, Post tick and pre tick events are before and after everything in the game has been dealt with, not just position rotation and onGround. So stating those as what's called before and after these events can confuse users.